src/Symfony/Component/HttpKernel/DataCollector/EventDataCollector.php

```diff
@@ -28,94 +28,108 @@
 {
-    private ?EventDispatcherInterface $dispatcher;
+    /** @var iterable<EventDispatcherInterface> */
+    private iterable $dispatchers;
     private ?RequestStack $requestStack;
     private ?Request $currentRequest = null;
 
-    public function __construct(EventDispatcherInterface $dispatcher = null, RequestStack $requestStack = null)
+    /**
+     * @param iterable<EventDispatcherInterface>|EventDispatcherInterface|null $dispatchers
+     */
+    public function __construct(iterable|EventDispatcherInterface|null $dispatchers = [], RequestStack $requestStack = null)
     {
-        $this->dispatcher = $dispatcher;
+        if (!is_iterable($dispatchers)) {
+            $dispatchers = $dispatchers ? ['event_dispatcher' => $dispatchers] : [];
+        }
+        $this->dispatchers = $dispatchers;
         $this->requestStack = $requestStack;
     }
 
     public function collect(Request $request, Response $response, \Throwable $exception = null): void
     {
         $this->currentRequest = $this->requestStack && $this->requestStack->getMainRequest() !== $request ? $request : null;
-        $this->data = [
-            'called_listeners' => [],
-            'not_called_listeners' => [],
-            'orphaned_events' => [],
-        ];
+        $this->data = [];
     }
 
     public function reset(): void
     {
         $this->data = [];
 
-        if ($this->dispatcher instanceof ResetInterface) {
-            $this->dispatcher->reset();
+        foreach ($this->dispatchers as $dispatcher) {
+            if ($dispatcher instanceof ResetInterface) {
+                $dispatcher->reset();
+            }
         }
     }
 
     public function lateCollect(): void
     {
-        if ($this->dispatcher instanceof TraceableEventDispatcher) {
-            $this->setCalledListeners($this->dispatcher->getCalledListeners($this->currentRequest));
-            $this->setNotCalledListeners($this->dispatcher->getNotCalledListeners($this->currentRequest));
-            $this->setOrphanedEvents($this->dispatcher->getOrphanedEvents($this->currentRequest));
+        foreach ($this->dispatchers as $name => $dispatcher) {
+            if (!$dispatcher instanceof TraceableEventDispatcher) {
+                continue;
+            }
+
+            $this->setCalledListeners($dispatcher->getCalledListeners($this->currentRequest), $name);
+            $this->setNotCalledListeners($dispatcher->getNotCalledListeners($this->currentRequest), $name);
+            $this->setOrphanedEvents($dispatcher->getOrphanedEvents($this->currentRequest), $name);
         }
 
         $this->data = $this->cloneVar($this->data);
     }
 
+    public function getData(): array|Data
+    {
+        return $this->data;
+    }
+
     /**
      * @see TraceableEventDispatcher
      */
-    public function setCalledListeners(array $listeners): void
+    public function setCalledListeners(array $listeners, string $dispatcher = 'event_dispatcher'): void
```

Comment:
I'm no a fan of adding the default name to the signature
then, in the implementation, what about returning the merged array of all called listeners when `$dispatcher` is null?
It'd make sense to me, both from a BC and from a feature POV.
(same idea for the other methods of course.)

Reply:
> what about returning the merged array of all called listeners when `$dispatcher` is null?

Not sure about that given https://github.com/symfony/symfony/pull/49151#issuecomment-1408482700

Also how would the setters behave in that case?